src/OpenTelemetry.Exporter.Zipkin/Implementation/ZipkinActivityConversionExtensions.cs

```diff
@@ -176,41 +208,7 @@ private static bool ProcessActivityEvents(ref PooledList<ZipkinAnnotation> annot
             return true;
         }
 
-        private static bool ProcessTags(ref AttributeEnumerationState state, KeyValuePair<string, object> attribute)
-        {
-            string key = attribute.Key;
-            string strVal = attribute.Value as string;
-
-            if (strVal != null)
-            {
-                if (RemoteEndpointServiceNameKeyResolutionDictionary.TryGetValue(key, out int priority)
-                    && (state.RemoteEndpointServiceName == null || priority < state.RemoteEndpointServiceNamePriority))
-                {
-                    state.RemoteEndpointServiceName = strVal;
-                    state.RemoteEndpointServiceNamePriority = priority;
-                }
-                else if (key == Resource.ServiceNameKey)
-                {
-                    state.ServiceName = strVal;
-                }
-                else if (key == Resource.ServiceNamespaceKey)
-                {
-                    state.ServiceNamespace = strVal;
-                }
-                else
-                {
-                    PooledList<KeyValuePair<string, object>>.Add(ref state.Tags, new KeyValuePair<string, object>(key, strVal));
-                }
-            }
-            else
-            {
-                PooledList<KeyValuePair<string, object>>.Add(ref state.Tags, new KeyValuePair<string, object>(key, strVal));
```

Comment:
I think all you need to do is fix the issue is change this line from...
`PooledList<KeyValuePair<string, object>>.Add(ref state.Tags, new KeyValuePair<string, object>(key, strVal));`
to
`PooledList<KeyValuePair<string, object>>.Add(ref state.Tags, new KeyValuePair<string, object>(key, attribute.Value));`

Reply:
that would solve, but we would have to keep the null chck in strval. In the end we simplified, no?

Reply:
and, the other question, should we check if attribute.value is null before adding?

Reply:
Nulls really shouldn't be possible. I think the spec says if you add a null, it is like removing the tag. The Activity version was bugged initially, but I think it should be working correctly now. Can you add a test and see if they are correctly eliminated?

Reply:
I had to add a condition. if we just try without doing that, it was inserting null values.

Reply:
Yeah, @CodeBlanch is right: per OTel spec the key is to be removed if the value is null so it shouldn't reach the exporter. Side note: we export JSON but IIRC Zipkin proto `tags` is a `map<string, string>` that shouldn't accept `null` as values anyway.

Reply:
changed the order to prevent two checks. about the null values, probably we will have to check the instrumentation(?)

Reply:
correct: just making the point that in practice it shouldn't reach the exporter if it does it is a bug.